apps/mobile/src/components/amount-field/amount-field.utils.ts

```diff
@@ -29,15 +29,17 @@ interface CalculateSecondaryValueParams {
   value: string;
   mode: InputCurrencyMode;
   marketData: MarketData;
+  assetDecimals?: number;
 }
 
 export function calculateSecondaryValue({
   value,
   mode,
   marketData,
+  assetDecimals,
 }: CalculateSecondaryValueParams) {
   const numericValue = Number(value);
-  const baseAmount = createMoneyFromDecimal(numericValue, marketData.pair.base);
+  const baseAmount = createMoneyFromDecimal(numericValue, marketData.pair.base, assetDecimals);
```

Comment:
Maybe we can avoid drilling down `assetDecimals` if we do a lookup of `asset` at this level where it's needed? 🤔

Reply:
amount field is a general purpose component, it shouldn't depend on any one feature's context.